File: src/main/java/de/tectoast/emolga/commands/draft/RoundCommand.java
package de.tectoast.emolga.commands.draft;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.emolga.utils.draft.Draft;

import java.util.Optional;

public class RoundCommand extends Command {
    public RoundCommand() {
        super("round", "Zeigt die Runde des derzeitigen Drafts an", CommandCategory.Draft);
        setArgumentTemplate(ArgumentManagerTemplate.noArgs());
    }

    @Override
    public void process(GuildCommandEvent e) {
        Optional<Draft> op = Draft.drafts.stream().filter(d -> d.tc.getId().equals(e.getChannel().getId())).findFirst();
        if (op.isEmpty()) {
            e.reply("In diesem Textchannel findet derzeit kein Draft statt!");
            return;
        }
        e.reply("Der Draft ist in Runde " + op.get().round + "!");
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/draft/RoundCommand.kt
package de.tectoast.emolga.commands.draft

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.emolga.utils.json.emolga.draft.League

object RoundCommand : Command("round", "Zeigt die Runde des derzeitigen Drafts an", CommandCategory.Draft) {
    init {
        argumentTemplate = ArgumentManagerTemplate.noArgs()
    }

    override suspend fun process(e: GuildCommandEvent) {
        e.reply(League.onlyChannel(e.textChannel.idLong)?.let { "Der Draft ist in Runde ${it.round}!" }
            ?: "Es läuft zurzeit kein Draft in diesem Channel!")
    }
}
